streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java

```diff
@@ -0,0 +1,531 @@
+            testMetricByName(listMetricTask, COMMIT_LATENCY_AVG, 5);
+            testMetricByName(listMetricTask, COMMIT_LATENCY_MAX, 5);
+            testMetricByName(listMetricTask, COMMIT_RATE, 5);
+            testMetricByName(listMetricTask, COMMIT_TOTAL, 5);
+            testMetricByName(listMetricTask, RECORD_LATENESS_AVG, 4);
+            testMetricByName(listMetricTask, RECORD_LATENESS_MAX, 4);
+            return true;
+        } catch (final Throwable e) {
+            errorMessage.append(e.getMessage());
+            return false;
+        }
+    }
+
+    private boolean testProcessorMetric(final StringBuilder errorMessage) {
+        errorMessage.setLength(0);
+        try {
+            final List<Metric> listMetricProcessor = new ArrayList<Metric>(kafkaStreams.metrics().values()).stream().filter(m -> m.metricName().group().equals(STREAM_PROCESSOR_NODE_METRICS)).collect(Collectors.toList());
+            testMetricByName(listMetricProcessor, PROCESS_LATENCY_AVG, 18);
+            testMetricByName(listMetricProcessor, PROCESS_LATENCY_MAX, 18);
+            testMetricByName(listMetricProcessor, PUNCTUATE_LATENCY_AVG, 18);
+            testMetricByName(listMetricProcessor, PUNCTUATE_LATENCY_MAX, 18);
+            testMetricByName(listMetricProcessor, CREATE_LATENCY_AVG, 18);
+            testMetricByName(listMetricProcessor, CREATE_LATENCY_MAX, 18);
+            testMetricByName(listMetricProcessor, DESTROY_LATENCY_AVG, 18);
+            testMetricByName(listMetricProcessor, DESTROY_LATENCY_MAX, 18);
+            testMetricByName(listMetricProcessor, PROCESS_RATE, 18);
+            testMetricByName(listMetricProcessor, PROCESS_TOTAL, 18);
+            testMetricByName(listMetricProcessor, PUNCTUATE_RATE, 18);
+            testMetricByName(listMetricProcessor, PUNCTUATE_TOTAL, 18);
+            testMetricByName(listMetricProcessor, CREATE_RATE, 18);
+            testMetricByName(listMetricProcessor, CREATE_TOTAL, 18);
+            testMetricByName(listMetricProcessor, DESTROY_RATE, 18);
+            testMetricByName(listMetricProcessor, DESTROY_TOTAL, 18);
+            testMetricByName(listMetricProcessor, FORWARD_TOTAL, 18);
+            return true;
+        } catch (final Throwable e) {
+            errorMessage.append(e.getMessage());
+            return false;
+        }
+    }
+
+    private boolean testStoreMetricWindow(final StringBuilder errorMessage) {
+        errorMessage.setLength(0);
+        try {
+            final List<Metric> listMetricStore = new ArrayList<Metric>(kafkaStreams.metrics().values()).stream()
+                    .filter(m -> m.metricName().group().equals(STREAM_STORE_WINDOW_ROCKSDB_STATE_METRICS))
+                    .collect(Collectors.toList());
+            testMetricByName(listMetricStore, PUT_LATENCY_AVG, 2);
+            testMetricByName(listMetricStore, PUT_LATENCY_MAX, 2);
+            testMetricByName(listMetricStore, PUT_IF_ABSENT_LATENCY_AVG, 0);
```

Comment:
Do I read this correctly: it's verifying that we have 0 metrics registered for `PUT_IF_ABSENT_LATENCY_AVG`?

Reply:
That's what i found after starting the app. I don't know the required "given" to got this type of metrics :)

Reply:
For window / session stores, there's no putIfAbsent function and hence no metrics would be registered.

Reply:
i remove it or let the test on 0 ?